test/socket.io.test-d.ts

```diff
@@ -0,0 +1,211 @@
+"use strict";
+import { DefaultEventsMap, Server, Socket } from "..";
+import { createServer } from "http";
+import { expectError, expectType } from "tsd";
+import "./support/util";
+
+describe("server", () => {
+  describe("no event map", () => {
+    describe("on", () => {
```

Comment:
I wrote this test in the same style as Mocha unit tests (`describe` / `it`), but these are not actually run. Perhaps it could make sense to refactor to some other structure, which might be less confusing.

Reply:
Good idea! Done :)